doc/source/whatsnew/v0.24.0.rst

```diff
@@ -1195,8 +1195,6 @@ Other API Changes
 - :class:`pandas.io.formats.style.Styler` supports a ``number-format`` property when using :meth:`~pandas.io.formats.style.Styler.to_excel` (:issue:`22015`)
 - :meth:`DataFrame.corr` and :meth:`Series.corr` now raise a ``ValueError`` along with a helpful error message instead of a ``KeyError`` when supplied with an invalid method (:issue:`22298`)
 - :meth:`shift` will now always return a copy, instead of the previous behaviour of returning self when shifting by 0 (:issue:`22397`)
-- :meth:`DataFrame.set_index` now gives a better (and less frequent) KeyError, raises a ``ValueError`` for incorrect types,
```

Comment:
leave this, it IS in 0.24.0

Reply:
You asked for a revert, which is exactly what I did, and where this came from.

Reply:
correct, but 0.24.0 release notes are already out there. it IS there in that version, so leave this comment.

Reply:
@jreback you are correct, but, from a practical point of view: a majority of users looking at this will be not be using 0.24.0, but something above that, so for those it is more informative to not contain that.

Reply:
I don't see the distinction. This exists currently in 0.24.0. Removing the note is a no-no. (if you want to add that it is reverted in 0.24.1, ok), but the note stays.

Reply:
ok, not going to fight over this one :)

Reply:
> You asked for a revert, which is exactly what I did, and where this came from.

I guess I should have added more explicitly that I re-added the lines in question at the same time as I made that comment. I was just explaining why it had been removed.